types/node/globals.d.ts

```diff
@@ -64,17 +64,21 @@ interface AbortSignal extends EventTarget {
     readonly aborted: boolean;
 }
 
-declare var AbortController: {
-    prototype: AbortController;
-    new(): AbortController;
-};
-
-declare var AbortSignal: {
-    prototype: AbortSignal;
-    new(): AbortSignal;
-    // TODO: Add abort() static
-    timeout(milliseconds: number): AbortSignal;
-};
+declare var AbortController: typeof globalThis extends {onmessage: any; AbortController: infer T}
+    ? T
+    : {
+        prototype: AbortController;
+        new(): AbortController;
+    };
+
+declare var AbortSignal: typeof globalThis extends {onmessage: any; AbortSignal: infer T}
```

Comment:
Were these just missed in the original PR? Or are they new in 4.9’s lib.dom.d.ts?

Reply:
I can only answer the second part: AbortSignal is not new in 4.9, but it changed to add `timeout` (and once this is fixed, I'm going to add `abort` in 4.9 as well in a few days).

Reply:
From discussion with @andrewbranch, it seems that @thw0rted answered the first question earlier in the typescript discord: it was indeed missed in the original PR, because it wasn't tested until now.

Reply:
@sandersn I'm also curious as to the origin of this pattern, AFAIC, I traced this pattern back to https://github.com/DefinitelyTyped/DefinitelyTyped/issues/34960#issuecomment-576906058 by @forivall.

Reply:
I got it from @thw0rted, where it was the first I had seen it. But it sounds like you've already done more detective work than I have.

I wish it wasn't necessary but it does help the types keep working even when certain globals are not defined.

Reply:
I can't take credit -- if you look back at the original "remove DOM typings" PR, I link to a previous change to the Node typings that already used the same construct.  I agree that it's not the great but unfortunately efforts to provide first-class language support for this kind of thing seem to have stalled out repeatedly over the years.

Reply:
Conditional types *are* first-class support, for lots of kludgey things. =P

Reply:
@thw0rted, I appreciate the clarity, but from my end, I am *explicitly* trying to look at the history of this to determine how stable it would be down the road for different use cases.

If there should be first-class support for this pattern or some other feature as a special case where module import order would augment the properties of `globalThis`, this needs more work on TypeScript's end, at least test cases to ensure the behaviour does not break.

Reply:
@thw0rted for reference, the earliest mention on a pull request was in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56163#discussion_r732279580 from last year, but the earliest comment I found was the above.

I agree with your points, and I think TypeScript needs to settle on a way to define `globalThis` properties within modules and there needs to be mechanisms to model this behaviour at runtime when transpiling across module formats that would be deterministic.